packages/containers/src/ObjectViewer/ObjectViewer.container.js

```diff
@@ -0,0 +1,92 @@
+import React, { PropTypes } from 'react';
+import Immutable, { List, Map } from 'immutable';
+import get from 'lodash/get';
+
+import { ObjectViewer as Component } from 'react-talend-components';
+import { statePropTypes, stateWillMount } from '../state';
+
+export const DEFAULT_STATE = new Map({
+	edited: new List(),  // Array of JSONPath
+	opened: new List(),  // Array of JSONPath
+	modified: new Map(),  // Store the onChange
+});
+
+class ObjectViewer extends React.Component {
+	static displayName = 'CMFContainer(ObjectViewer)';
+	static propTypes = {
+		id: PropTypes.string,
+		data: get(Component, 'propTypes.data'),
+		displayMode: get(Component, 'propTypes.displayMode'),
+		onSubmit: get(Component, 'propTypes.onSubmit'),
+		...statePropTypes,
+	};
+
+	constructor(props) {
+		super(props);
+		this.onClick = this.onClick.bind(this);
+		this.onChange = this.onChange.bind(this);
+	}
+
+	componentWillMount() {
+		stateWillMount(this.props);
+	}
+
+	onClick(event, data) {
+		const opened = this.props.state.get('opened');
+		const edited = this.props.state.get('edited');
+		let newState;
+		if (data.isOpened) {
+			// we should close the tree
+			newState = this.props.state.set(
```

Comment:
I guess a function would make more sense than a comment. 
Plus you could reduce repeated code.